content/5_Plat/Sweep_Line.mdx

```diff
@@ -162,8 +162,154 @@ For each point, we recalculate $\delta = \min(\delta, \delta_p)$, and update our
+by using a set (called 'active_segments' in the code below). When we hit the beginning point of a segment, we add it to the active set, and when we hit the ending point of a segment, we remove it from the active set.
+The operation of inserting or removing takes $\mathcal{O}(\log n)$.
+
+The active set of segments is ordered by $y$ coordinate. If two segments overlap, it means that they are adjacent in the set. So every time we insert or remove
+a segment we check if the adjacent segments overlap.
+
+Here is an animation of how it works:
+
+<video width="960" height="720" controls>
+	<source src="/animations/cowsteeplechase.mp4" type="video/mp4"/>
+</video>
+
+
+Time complexity: $\mathcal{O}(n\log n)$.
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+long long sweep_line_x;
+
+struct Point {
+	long long x, y, segment_idx;
+	bool operator<(const Point &oth) {
+		return x == oth.x ? y < oth.y : x < oth.x;
+	}
+};
+struct Segment {
+	long long a, b, x, y, idx;
+};
+
+double coordinate(Segment a) {
+	if (a.a == a.x) return a.b;
+	return a.b + (a.y - a.b) * (sweep_line_x - a.a) / (a.x - a.a);
+}
+
+int sign(long long x) {
+	if (x == 0) return 0;
+	else return x < 0 ? -1 : +1;
+}
+int operator*(Point p1, Point p2) { return sign(p1.x * p2.y - p1.y * p2.x); }
+Point operator-(Point p1, Point p2) {
+	Point p = {p1.x - p2.x, p1.y - p2.y};
+	return p;
+}
+bool intersect(Segment a, Segment b) {
+	Point p1 = {a.a, a.b}, q1 = {a.x, a.y}, p2 = {b.a, b.b}, q2 = {b.x, b.y};
+	return ((q2 - p1) * (q1 - p1)) * ((q1 - p1) * (p2 - p1)) >= 0 &&
+	       ((q1 - p2) * (q2 - p2)) * ((q2 - p2) * (p1 - p2)) >= 0;
+}
+
+bool operator<(Segment a, Segment b) {
+	return a.idx != b.idx && coordinate(a) <= coordinate(b);
+}
+bool operator==(Segment a, Segment b) { return a.idx == b.idx; }
+long long orientation(Point a, Point b, Point c) {
+	return (b.x - a.x) * (c.y - a.y) - (c.x - a.x) * (b.y - a.y);
+}
+
+ifstream in("cowjump.in");
+ofstream out("cowjump.out");
+
+int main() {
+	int n;
+	in >> n;
+	vector<Segment> segments;
+	vector<Point> events;
+	for (int i = 0; i < n; i++) {
+		int a, b, x, y;
+		in >> a >> b >> x >> y;
+
+		segments.push_back({a, b, x, y, i});
+		events.push_back({a, b, i});
+		events.push_back({x, y, i});
+	}
+
+	sort(events.begin(), events.end());
+
+	// Keep track of active segments
+	set<Segment> active_segments;
+	// The two overlapping segments
+	int first_segment, second_segment;
+	for (int i = 0; i < 2 * n; i++) {
+		first_segment = events[i].segment_idx;
+		sweep_line_x = events[i].x;
+		// Check if the point is the end or the beginning of a segment
+		auto it = active_segments.find(segments[first_segment]);
+
+		if (it != active_segments.end()) {
+			// ending segment
+			// Check the intersection of the segments above and below
```

Comment:
try to be consistent with capitalization of comments :D

Reply:
Alright